src/lib.rs

```diff
@@ -105,11 +198,21 @@ fn value_to_pyobject(value: Value) -> PyResult<PyObject> {
 fn pyobject_to_value<'v>(obj: PyObject, heap: &'v Heap) -> PyResult<Value<'v>> {
     Python::with_gil(|py| -> PyResult<Value<'v>> {
         let json = py.import_bound("json")?;
-        let json_str: String = json.getattr("dumps")?.call1((obj,))?.extract()?;
-        convert_anyhow_err(serde_to_starlark(
-            convert_serde_err(serde_json::from_str(&json_str))?,
-            heap,
-        ))
+
+        // Try JSON serialization first
+        match json.getattr("dumps")?.call1((obj.clone_ref(py),)) {
+            Ok(json_str) => {
+                let json_str: String = json_str.extract()?;
+                convert_anyhow_err(serde_to_starlark(
+                    convert_serde_err(serde_json::from_str(&json_str))?,
+                    heap,
+                ))
+            }
+            Err(_) => {
+                // If JSON serialization fails, wrap in PythonObjectValue
+                Ok(heap.alloc(PythonObjectValue { obj: obj }))
```

Comment:
This can be simplified. See https://rust-lang.github.io/rust-clippy/master/index.html#redundant_field_names

```suggestion
                Ok(heap.alloc(PythonObjectValue { obj }))
```